buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy

```diff
@@ -42,7 +42,7 @@ class ClassyclePlugin implements Plugin<Project> {
             def taskName = sourceSet.getTaskName('classycle', null)
             project.tasks.create(taskName, Classycle, { Classycle task ->
                 task.reportName = sourceSet.name
-                task.classesDir = sourceSet.output.classesDir
+                task.classesDirs = sourceSet.output.classesDirs.files.findAll { it.exists() }
```

Comment:
Don't we resolve the file collection here too early? This now happens at configuration time before any of the `classesDirs` exist, right?
I think this should happen during execution when iterating over the directories.